src/tests/Microsoft.PowerFx.Core.Tests.Shared/ExpressionTestCases/JSON.txt

```diff
@@ -133,3 +133,80 @@ Error({Kind:ErrorKind.Div0})
 // Flattening nested tables
 >> JSON([[1,2,3],[4,5],[6]], JSONFormat.FlattenValueTables)
 "[[1,2,3],[4,5],[6]]"
+
+>> JSON(ParseJSON("{}"))
+"{}"
+
+>> JSON(ParseJSON("[]"))
+"[]"
+
+>> JSON(ParseJSON("1"))
+"1"
+
+>> JSON(ParseJSON("1.77"))
+"1.77"
+
+>> JSON(ParseJSON("-871"))
+"-871"
+
+>> JSON(ParseJSON("""John"""))
+"""John"""
+
+>> JSON(ParseJSON("true"))
+"true"
+
+>> JSON(ParseJSON("false"))
+"false"
+
+>> JSON(ParseJSON("{""a"": 1}"))
+"{""a"":1}"
+
+>> JSON(ParseJSON("{""a"": ""x""}"))
+"{""a"":""x""}"
+
+>> JSON(ParseJSON("[1]"))
+"[1]"
+
+>> JSON(ParseJSON("{""a"": 1.5}"))
+"{""a"":1.5}"
+
+>> JSON(ParseJSON("[1.5]"))
+"[1.5]"
+
+>> JSON(ParseJSON("{""a"":[1]}"))
+"{""a"":[1]}"
+
+>> JSON(ParseJSON("[{""a"": -17}]"))
+"[{""a"":-17}]"
+
+>> JSON(ParseJSON("[true, false]"))
+"[true,false]"
+
+>> JSON(ParseJSON("[""True"", ""False""]"))
+"[""True"",""False""]"
+
+>> JSON(ParseJSON("   {  ""a"" :  1 }  "))
+"{""a"":1}"
+
+>> JSON(ParseJSON("{""a"": {""a"": 1}}"))
+"{""a"":{""a"":1}}"
+
+// Depth 21
+>> JSON(ParseJSON("{""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": {""a"": 1}}}}}}}}}}}}}}}}}}}}}"))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> JSON(ParseJSON("[[1]]"))
+"[[1]]"
+
+// Depth 21
+>> JSON(ParseJSON("[[[[[[[[[[[[[[[[[[[[[1]]]]]]]]]]]]]]]]]]]]]"))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> JSON(Decimal(ParseJSON("123456789012345.6789012345678")))
+"123456789012345.6789012345678"
+
+>> JSON(ParseJSON("123456789012345.6789012345678"))
+"123456789012345.6789012345678"
+
+>> JSON(ParseJSON("""\u0048\u0065\u006c\u006c\u006f"""))
```

Comment:
>> JSON(ParseJSON("""\u0048\u0065\u006c\u006c\u006f""")) [](http://example.com/codeflow?start=1&length=55)

Can you add a comment saying something like "Round-trip is not guaranteed - escaped characters that don't need escaping will not be re-escaped when JSON-ified".

Similar comment on the test on line 188 (that whitespaces are not round-tripped)